harmonica/forward/prisms_layer.py

```diff
@@ -0,0 +1,411 @@
+            passed as 2d array. Height(s) must be in meters.
+        """
+        if surface.shape != self.shape:
+            raise ValueError(
+                f"Invalid surface array with shape '{surface.shape}'. "
+                + "Its shape should be compatible with the coordinates "
+                + "of the layer of prisms."
+            )
+        if isinstance(reference, np.ndarray):
+            if reference.shape != self.shape:
+                raise ValueError(
+                    f"Invalid reference array with shape '{reference.shape}'. "
+                    + "Its shape should be compatible with the coordinates "
+                    + "of the layer of prisms."
+                )
+        else:
+            reference *= np.ones(self.shape)
+        top = surface.copy()
+        bottom = reference.copy()
+        reverse = surface < reference
+        top[reverse] = reference[reverse]
+        bottom[reverse] = surface[reverse]
+        self._obj.coords["top"] = (self.dims, top)
+        self._obj.coords["bottom"] = (self.dims, bottom)
+
+    def gravity(self, coordinates, field, density_name="density", **kwargs):
+        """
+        Computes the gravity generated by the layer of prisms
+
+        Uses :func:`harmonica.prism_gravity` for computing the gravity field
+        generated by the prisms of the layer.
+        The density of the prisms will be assigned from the ``data_var`` chosen
+        through the ``density_name`` argument.
+        Ignores the prisms which ``top`` or ``bottom`` boundaries are
+        ``np.nan``s.
+        All ``kwargs`` will be passed to :func:`harmonica.prism_gravity`.
+
+        Parameters
+        ----------
+        coordinates : list or 1d-array
+            List or array containing ``easting``, ``northing`` and ``upward``
+            of the computation points defined on a Cartesian coordinate system.
+            All coordinates should be in meters.
+        field : str
+            Gravitational field that wants to be computed.
+            The available fields are:
+            - Gravitational potential: ``potential``
+            - Downward acceleration: ``g_z``
+        density_name : str (optional)
+            Name of the property layer (or ``data_var`` of the
+            :class:`xarray.Dataset`) that will be used for the density of each
+            prism in the layer. Default to ``"density"``
+
+        Returns
+        -------
+        result : array
+            Gravitational field generated by the prisms on the computation
+            points in mGal.
+
+        See also
+        --------
+        harmonica.prism_gravity
+        """
+        boundaries = self.get_prisms()
+        # Get the mask for selecting only the boundaries of the prisms that
+        # have no nans on their top and bottom
+        nonans_boundaries = self._get_nonans_prisms_mask().ravel()
+        # Get density array
+        density = getattr(self._obj, density_name).values.ravel()
```

Comment:
```suggestion
        density = self._obj[density_name].values.ravel()
```
Works for density names that aren't valid Python variable names.